goosebit/updates/swdesc.py

```diff
@@ -72,7 +73,17 @@ async def parse_remote(url: str):
             return await parse_file(Path(f.name))
 
 
-def _sha1_hash_file(file_path: Path):
-    with file_path.open("rb") as f:
-        sha1_hash = hashlib.file_digest(f, "sha1")
+async def _sha1_hash_file(fileobj: AsyncFile):
+    last = await fileobj.tell()
+    await fileobj.seek(0)
+    sha1_hash = hashlib.sha1()
+    buf = bytearray(2**18)
+    view = memoryview(buf)
+    while True:
+        size = await fileobj.readinto(buf)
+        if size == 0:
+            break
+        sha1_hash.update(view[:size])
+
+    await fileobj.seek(last)
```

Comment:
Is there a reason to do this in this format?  Probably better on RAM, but even the worst server equipment should have enough ram to handle a few of these files at a time...

Reply:
This is basically a straight port of the stdlib's `hashlib.file_digest` function to use anyio instead of normal synchronous IO. IMO we should generally operate under the assumption that swu files may not fit into ram always, the upper limit on swu file sizes is rather high as well.